File: src/main/scala/com/mikemunhall/scalakatas/scala/FizzBuzz.scala
package scala.com.mikemunhall.scalakatas.scala

object FizzBuzz {

  /*def getResult(in: Int): String = {
    in match {
      case x if in % 15 == 0 => "fizzbuzz"
      case x if in % 3 == 0 => "fizz"
      case x if in % 5 == 0 => "buzz"
      case other => other.toString()
    }
  }*/

  def getResult(divisorA: Int, divisorB: Int)(testNum: Int): String = {
    (testNum % divisorA, testNum % divisorB) match {
      case (0, 0) => "fizzbuzz"
      case (0, _) => "fizz"
      case (_, 0) => "buzz"
      case (_, _) => testNum.toString
    }
  }
}

File: src/main/java/com/mikemunhall/katas/java/FizzBuzz.java
package com.mikemunhall.katas.java;


public class FizzBuzz {

    public static String getResult(Integer x) {
        if (x % 3 == 0 && x % 5 == 0) {
            return "fizzbuzz";
        } else if (x % 3 == 0) {
            return "fizz";
        } else if (x % 5 == 0) {
            return "buzz";
        } else {
            return x.toString();
        }
    }
}
